metricflow/sql/optimizer/rewriting_sub_query_reducer.py

```diff
@@ -612,10 +615,11 @@ def visit_select_statement_node(self, node: SqlSelectStatementNode) -> SqlQueryP
         # JOIN dim_listings c
         # ON a.listing_id = b.listing_id
 
-        assert len(node_with_reduced_parents.parent_nodes) == 1
```

Comment:
Was there a reason to remove this assertion?

Reply:
Yes - since CTEs are modeled as a parent node of a SELECT node, this assertion does not hold.